packages/aws-cdk-lib/aws-cognito/lib/user-pool-idps/oidc.ts

```diff
@@ -140,6 +136,9 @@ export class UserPoolIdentityProviderOidc extends UserPoolIdentityProviderBase {
       if (!Token.isUnresolved(name) && (name.length < 3 || name.length > 32)) {
         throw new Error(`Expected provider name to be between 3 and 32 characters, received ${name} (${name.length} characters)`);
       }
+      if (!name.match(/[^_\p{Z}][\p{L}\p{M}\p{S}\p{N}\p{P}][^_\p{Z}]+/)) {
```

Comment:
Because this regex is long and uses less-common character classes, what do you think about linking to the docs that reference the pattern?

```suggestion
      // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cognito-userpoolidentityprovider.html#cfn-cognito-userpoolidentityprovider-providername
      if (!name.match(/[^_\p{Z}][\p{L}\p{M}\p{S}\p{N}\p{P}][^_\p{Z}]+/)) {
```

Reply:
good shout 👍🏼

Reply:
```suggestion
      if (!name.match(/^[^_\p{Z}][\p{L}\p{M}\p{S}\p{N}\p{P}][^_\p{Z}]+$/u)) {
```

This will ensure the full string matches and that unicode mode is enabled.